src/ServiceInsight/Framework/Commands/ObservableCommand.cs

```diff
@@ -0,0 +1,62 @@
+namespace ServiceInsight.Framework.Commands
+{
+    using System;
+    using System.Reactive.Disposables;
+    using System.Reactive.Linq;
+    using System.Windows.Input;
+
+    class ObservableCommand : ICommand
+    {
+        private Action<object> action;
+        private IDisposable canExecuteSubscription;
+        private bool latest;
+        private bool isExecuting;
+
+        public ObservableCommand(IObservable<bool> canExecuteObservable, Action<object> action)
+        {
+            this.action = action;
+
+            canExecuteSubscription = canExecuteObservable
+                .Subscribe(b =>
+                {
+                    latest = b;
+                    RaiseCanExecuteChanged();
+                });
+        }
+
+        public event EventHandler CanExecuteChanged;
+
+        public bool CanExecute(object parameter) => !isExecuting && latest;
+
+        public void Execute(object parameter)
+        {
+            using (StartExecuting())
+            {
+                action(parameter);
+            }
+        }
+
+        public void RaiseCanExecuteChanged()
+        {
+            CanExecuteChanged?.Invoke(this, EventArgs.Empty);
+        }
+
+        public void Dispose()
```

Comment:
I just realised that this class does not implement `IDisposable` and this method has no callers, so it's redundant. Should we remove it, or should this class be `IDisposable`?